runtime/autoload/health/provider.vim

```diff
@@ -437,54 +328,17 @@ function! s:check_python(version) abort
       call health#report_info(printf('pynvim version: %s', current))
     endif
 
-    if s:is_bad_response(current)
-      call health#report_error(
-        \ "pynvim is not installed.\nError: ".current,
-        \ ['Run in shell: '. pip .' install pynvim'])
-    endif
-
     if s:is_bad_response(latest)
-      call health#report_warn('Could not contact PyPI to get latest version.')
+      call health#report_warn('Could not contact PyPI to check if pynvim is latest version.')
       call health#report_error('HTTP request failed: '.latest)
     elseif s:is_bad_response(status)
       call health#report_warn(printf('Latest pynvim is NOT installed: %s', latest))
-    elseif !s:is_bad_response(current)
+    else
       call health#report_ok(printf('Latest pynvim is installed.'))
     endif
   endif
 endfunction
 
-" Check if pyenv is available and a valid pyenv root can be found, then return
-" their respective paths. If either of those is invalid, return two empty
-" strings, effectivly ignoring pyenv.
-function! s:check_for_pyenv() abort
```

Comment:
`pyenv` has no useful information to contribute regarding Python provider detection (apart from the possible error output captured in `provider/pythonx`), and as I've argued previously, I don't think it's worth it to maintain troubleshooting code for a third-party tool for no particular reason.